kpatch-build/create-diff-object.c

```diff
@@ -922,6 +1020,9 @@ static void kpatch_replace_sections_syms(struct kpatch_elf *kelf)
 				continue;
 			}
 
+#ifdef __powerpc__
+			add_off = 0;
+#else
```

Comment:
I don't see how this would work for relative relocations, like PPC64_REL24.  In those cases I think the logic would be similar to what x86 does in the R_X86_64_PC32 case, because the address offset would be relative to the instruction address.  Or am I missing something?

Thankfully you wouldn't need rela_insn(), because that's just needed for determining the instruction length.  ppc has a fixed instruction size so it wouldn't need to disassemble.